components/RegisterISCN.vue

```diff
@@ -70,29 +70,32 @@ const iscnFee = ref(new BigNumber(0))
 const iscnGasFee = ref('0')
 const uploadStatus = ref('')
 const error = ref('')
-const emit = defineEmits(['handleSubmit', 'submit'])
+const emit = defineEmits(['handleSubmit', 'submit', 'formValidChange'])
 
 const totalFee = computed(() => {
   return iscnFee.value || new BigNumber(0)
 })
 
 const { payload } = useISCN(iscnData)
 
-const isFormValid = computed(() => {
+const formError = computed(() => {
   const requiredFields = {
     title: !!iscnData.value.title,
     description: !!iscnData.value.description,
     authorName: !!iscnData.value.author.name,
     contentUrl: !!iscnData.value.contentFingerprints.some(f => !!f.url)
   }
 
-  error.value = Object.entries(requiredFields)
-    .find(([_, isValid]) => !isValid)?.[0]
-    ?.toUpperCase() || ''
-
-  return Object.values(requiredFields).every(Boolean)
+  return Object.entries(requiredFields)
+    .find(([_, isValid]) => !isValid)?.[0]?.toUpperCase() || ''
 })
 
+const isFormValid = computed(() => !formError.value)
+
+watch(isFormValid, (val) => {
+  emit('formValidChange', val)
+}, { immediate: false })
+
```

Comment:
I think we can expose `isFormValid` as `defineModel` and use `v-model:valid` outside